### ISSUE TITLE
Domain adaptation - Adversarial Dropout Regularization - Keras implementation

### ISSUE THREAD

**Issue by jecaJeca** [keras-team/keras#10719]
Hello,

I am trying to implement in Keras domain adaptation method Adversarial Dropout Regularization from the paper https://arxiv.org/abs/1711.01575
Critic is classifier with dropouts and it is updated based on losses from more then one batch. Idea is to get two outputs for one target batch (dropout is included in test phase ), and measure KL divergence between outputs (they will not be same because of dropouts). Additionally to this value, critic loss is also depended on source batch loss. The final loss value is subtraction of source batch loss and discrepancy between outputs of target batch. (Eq 3 and 4, page 4), 

I don't know how to implement this  loss, due it depends on outputs of three batches (two target and one source). 

I tried to make one batch consisted of one source batch and two target batches (same target batch), and then in custom loss to separate y_pred and calculate measures. Here is my custom loss : 

    def custom_loss(self,y_true,y_pred):
        pred_source = y_pred[:self.batchSize,:]
        pred_tgt_1 = y_pred[self.batchSize+1:2*self.batchSize,:]
        pred_tgt_2 = y_pred[2*self.batchSize+1:,:]
        discp = discrepancy(pred_tgt_1,pred_tgt_2)
        loss = keras.losses.categorical_crossentropy(y_true[:self.batchSize],pred_source)

        return  loss - discp

I got error _Incompatible shapes: [64] vs. [192]_ (batch size is 64 ).

Dropout layers are added like this : 
`cl = layers.Dropout(0.5)(cl,training=True)`

Can someone help me with this?
